traefik/templates/service-internal.yaml

```diff
@@ -0,0 +1,123 @@
+{{- if .Values.service.internal.enabled -}}
+
+{{ $tcpPorts := dict }}
+{{ $udpPorts := dict }}
+{{- range $name, $config := .Values.ports }}
+  {{- if $config.exposeInternal }}
+    {{- if eq (toString $config.protocol) "UDP" }}
+      {{ $_ := set $udpPorts $name $config }}
+    {{- else }}
+      {{ $_ := set $tcpPorts $name $config }}
+    {{- end }}
+  {{- end }}
+{{- end }}
+
+{{- if or (gt (len $tcpPorts) 0) (gt (len $udpPorts) 0) }}
+apiVersion: v1
+kind: List
```

Comment:
What is this List Resource for?

Reply:
Copy/pasted from the main Service template file: creating one or more objects from a single template file, while allowing unit tests to access those objects individually, querying a specific index from that List items array.